<launch>

  <!-- Arguments -->
  <arg name="robot_model_puppet"                default="vx300s"/>

  <arg name="base_link_puppet"                  default="base_link"/>

  <arg name="puppet_modes_left"                 default="$(find aloha)/config/puppet_modes_left.yaml"/>

  <arg name="launch_driver"                     default="true"/>
  <arg name="use_sim"                           default="true"/>
  <arg name="use_rviz"                          default="true"/>


  <arg name="robot_name_puppet_left"            value="puppet_left"/>


  <!-- Puppet Left Robot -->
  <include if="$(arg launch_driver)" file="$(find interbotix_xsarm_control)/launch/xsarm_control.launch">
    <arg name="robot_model"                       value="$(arg robot_model_puppet)"/>
    <arg name="robot_name"                        value="$(arg robot_name_puppet_left)"/>
    <arg name="base_link_frame"                   value="$(arg base_link_puppet)"/>
    <arg name="use_world_frame"                   value="false"/>
    <arg name="use_rviz"                          value="false"/> 
    <arg name="mode_configs"                      value="$(arg puppet_modes_left)"/>
    <arg name="use_sim"                           value="$(arg use_sim)"/>
  </include>


  <node
    name="puppet_left_transform_broadcaster"
    pkg="tf2_ros"
    type="static_transform_publisher"
    args="0.0 0.0 0 0 0 0 /world /$(arg robot_name_puppet_left)/base_link"/>

  <node if="$(arg use_rviz)" pkg="rviz" type="rviz" name="rviz" args="-d /home/dexweaver/Documents/GitHub/cucumber-harvesting/rviz/rviz.rviz"/>

</launch>
